src/mpol/datasets.py

```diff
@@ -101,7 +101,10 @@ def add_mask(
             device (torch.device) : the desired device of the dataset. If ``None``, defalts to current device.
         """
 
-        new_2D_mask = torch.tensor(mask, device=device)
+        if torch.is_tensor(mask):
+            new_2D_mask = mask.clone().detach().to(device)
+        else:
+            new_2D_mask = torch.tensor(mask, device=device)
```

Comment:
Can be simply replaced with: `new_2D_mask = torch.Tensor(mask).detach().to(device)`